src/scene/particle-system/particle-emitter.js

```diff
@@ -1194,16 +1193,19 @@ class ParticleEmitter {
         this.colorParam?.destroy();
         this.colorParam = null;
 
-        this.vertexBuffer?.destroy();
         this.vertexBuffer = undefined; // we are testing if vb is undefined in some code, no idea why
-
-        this.indexBuffer?.destroy();
         this.indexBuffer = undefined;
 
+        const wasVisible = this.meshInstance?.visible ?? true;
```

Comment:
Wait...isn't `wasVisible` declared at line 584?

Reply:
It was. And now it's declared here as well.
Old code was not destroying mesh instance at all .. and so was reading it from it. Now I destroy it, and return it from there.